ethereum/api/src/main/java/org/hyperledger/besu/ethereum/api/jsonrpc/methods/DebugJsonRpcMethods.java

```diff
@@ -77,25 +80,40 @@ protected RpcApi getApiGroup() {
             blockchainQueries.getBlockchain(),
             blockchainQueries.getWorldStateArchive());
 
-    return mapOf(
-        new DebugTraceTransaction(blockchainQueries, new TransactionTracer(blockReplay)),
-        new DebugAccountRange(blockchainQueries),
-        new DebugStorageRangeAt(blockchainQueries, blockReplay),
-        new DebugMetrics(metricsSystem),
-        new DebugTraceBlock(
-            () -> new BlockTracer(blockReplay),
-            ScheduleBasedBlockHeaderFunctions.create(protocolSchedule),
-            blockchainQueries),
-        new DebugTraceBlockByNumber(() -> new BlockTracer(blockReplay), blockchainQueries),
-        new DebugTraceBlockByHash(() -> new BlockTracer(blockReplay)),
-        new DebugBatchSendRawTransaction(transactionPool),
-        new DebugGetBadBlocks(blockchainQueries, protocolSchedule, blockResult),
-        new DebugStandardTraceBlockToFile(
-            () -> new TransactionTracer(blockReplay), blockchainQueries, dataDir),
-        new DebugStandardTraceBadBlockToFile(
-            () -> new TransactionTracer(blockReplay),
-            blockchainQueries,
-            protocolSchedule,
-            dataDir));
+    final Map<String, JsonRpcMethod> debugJsonRpcMethods =
+        mapOf(
+            new DebugTraceTransaction(blockchainQueries, new TransactionTracer(blockReplay)),
+            new DebugAccountRange(blockchainQueries),
+            new DebugStorageRangeAt(blockchainQueries, blockReplay),
+            new DebugMetrics(metricsSystem),
+            new DebugTraceBlock(
+                () -> new BlockTracer(blockReplay),
+                ScheduleBasedBlockHeaderFunctions.create(protocolSchedule),
+                blockchainQueries),
+            new DebugTraceBlockByNumber(() -> new BlockTracer(blockReplay), blockchainQueries),
+            new DebugTraceBlockByHash(() -> new BlockTracer(blockReplay)),
+            new DebugBatchSendRawTransaction(transactionPool),
+            new DebugGetBadBlocks(blockchainQueries, protocolSchedule, blockResult),
+            new DebugStandardTraceBlockToFile(
+                () -> new TransactionTracer(blockReplay), blockchainQueries, dataDir),
+            new DebugStandardTraceBadBlockToFile(
+                () -> new TransactionTracer(blockReplay),
+                blockchainQueries,
+                protocolSchedule,
+                dataDir));
+    protocolSchedule
+        .getChainId()
+        .ifPresent(
+            chainId -> {
+              final DebugFaucet debugFaucet =
+                  new DebugFaucet(
+                      blockchainQueries,
+                      transactionPool,
+                      chainId,
+                      ExperimentalFeatures.faucetPrivateKey,
+                      Wei.fromEth(ExperimentalFeatures.faucetValue));
+              debugJsonRpcMethods.put(debugFaucet.getName(), debugFaucet);
+            });
```

Comment:
How come this is conditional on chainId? Could we instead make the private key field in the experimental options class `Optional<String>` and make this operation conditional on that?